binaries/daemon/src/lib.rs

```diff
@@ -251,21 +251,57 @@ impl Daemon {
             None => None,
         };
 
-        let zenoh_config = match std::env::var(zenoh::Config::DEFAULT_CONFIG_PATH_ENV) {
-            Ok(path) => zenoh::Config::from_file(&path)
-                .map_err(|e| eyre!(e))
-                .wrap_err_with(|| format!("failed to read zenoh config from {path}"))?,
-            Err(std::env::VarError::NotPresent) => zenoh::Config::default(),
+        let zenoh_session = match std::env::var(zenoh::Config::DEFAULT_CONFIG_PATH_ENV) {
+            Ok(path) => {
+                let zenoh_config = zenoh::Config::from_file(&path)
+                    .map_err(|e| eyre!(e))
+                    .wrap_err_with(|| format!("failed to read zenoh config from {path}"))?;
+                zenoh::open(zenoh_config)
+                    .await
+                    .map_err(|e| eyre!(e))
+                    .context("failed to open zenoh session")?
+            }
+            Err(std::env::VarError::NotPresent) => {
+                let mut zenoh_config = zenoh::Config::default();
+
+                if let Some(addr) = coordinator_addr {
+                    zenoh_config
+                        .insert_json5(
+                            "connect/endpoints",
+                            &format!(r#"["tcp/{}:5456"]"#, addr.ip()),
+                        )
+                        .unwrap();
```

Comment:
As discussed in our meeting, I'm not sure if this is a good default in all cases. You might not have a zenoh router running or the zenoh router could be running on a different machine (to load-balance).

I'm ok with going ahead with this change for now, but I think we should look for some better default (e.g. only fill endpoints if coordinator IP is not 127.0.0.1, start zenoh rounter automatically by coordinator, etc).

Reply:
So, this is not about zenoh router but zenoh peers. We are basically connecting the daemons together without routers. 

If we don't modify "connect/endpoints", the default is to listen on `tcp:[::]:0` which can be annoying to retrieve.

But this does not depend on spawning a zenoh router.